src/traces/bar/style.js

```diff
@@ -35,34 +35,55 @@ module.exports = function style(gd, cd) {
 
     s.selectAll('g.points').each(function(d) {
         var sel = d3.select(this);
-        var pts = sel.selectAll('path');
-        var txs = sel.selectAll('text');
         var trace = d[0].trace;
+        stylePoints(sel, trace, gd);
+    });
 
-        Drawing.pointStyle(pts, trace, gd);
-        Drawing.selectedPointStyle(pts, trace);
+    Registry.getComponentMethod('errorbars', 'style')(s);
+}
 
-        txs.each(function(d) {
-            var tx = d3.select(this);
-            var textFont;
+function stylePoints(sel, trace, gd) {
+    var pts = sel.selectAll('path');
+    var txs = sel.selectAll('text');
 
-            if(tx.classed('bartext-inside')) {
-                textFont = trace.insidetextfont;
-            } else if(tx.classed('bartext-outside')) {
-                textFont = trace.outsidetextfont;
-            }
-            if(!textFont) textFont = trace.textfont;
+    Drawing.pointStyle(pts, trace, gd);
+    Drawing.selectedPointStyle(pts, trace);
 
-            function cast(k) {
-                var cont = textFont[k];
-                return Array.isArray(cont) ? cont[d.i] : cont;
-            }
+    txs.each(function(d) {
+        var tx = d3.select(this);
+        var textFont;
 
-            Drawing.font(tx, cast('family'), cast('size'), cast('color'));
-        });
+        if(tx.classed('bartext-inside')) {
+            textFont = trace.insidetextfont;
+        } else if(tx.classed('bartext-outside')) {
+            textFont = trace.outsidetextfont;
+        }
+        if(!textFont) textFont = trace.textfont;
+
+        function cast(k) {
+            var cont = textFont[k];
+            return Array.isArray(cont) ? cont[d.i] : cont;
+        }
 
-        Drawing.selectedTextStyle(txs, trace);
+        Drawing.font(tx, cast('family'), cast('size'), cast('color'));
     });
 
-    Registry.getComponentMethod('errorbars', 'style')(s);
+    Drawing.selectedTextStyle(txs, trace);
+}
+
+function styleOnSelect(gd, cd) {
+    var s = cd[0].node3;
+    var trace = cd[0].trace;
+
+    if(trace.selectedpoints) {
+        Drawing.selectedPointStyle(s.selectAll('path'), trace, gd);
+        Drawing.selectedPointStyle(s.selectAll('text'), trace, gd);
+    } else {
+        stylePoints(s, trace, gd);
```

Comment:
Why do we need this `else`? I don't see a corresponding case in the previous code...

BTW, I'm hesitant to suggest this after the `.select().select()` data-mangling mess, but does it work to do `Drawing.selectedPointStyle(s.selectAll('path,text'), trace, gd)`? Is that any faster?

Reply:
Ah oops, this should be `Drawing.selectedTextStyle(s.selectAll('text'), trace, gd)`. I'll add a test.

Reply:
> Why do we need this else? I

To get back to the original state after double-click. `Drawing.selectedPointStyle` and `Drawing.selectedTextStyle` only handle `selected` / `unselected` styling off a "base" state given by `stylePoints`.

Reply:
> I'll add a test.

added in https://github.com/plotly/plotly.js/pull/2583/commits/01df2d2179d0a117fc17deee288047a637cffab0